aio/src/app/custom-elements/events/events.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+
+import { EventsService } from './events.service';
+
+export interface Event {
+  endDate: string;      // "2020-05-27",
+  address: string;      // "Oslo, Norway",
+  title: string;        // "ngVikings",
+  linkUrl: string;      // "https://ngvikings.org/",
+  name: string;          // "ngVikings",
+  info: string;          // "May 25-26 conference, 27 workshops, 2020"
+}
+
+@Component({
+  selector: 'aio-events',
+  templateUrl: 'events.component.html'
+})
+export class EventComponent implements OnInit {
+
+  prevEvents: Event[];
+  upcomingEvents: Event[];
+
+  constructor( private eventsService: EventsService ) { }
+
+  ngOnInit() {
+    this.eventsService.events.subscribe(( events: Event[] ) => {
+      this.prevEvents = events.filter( event => {
+        return new Date(event.endDate).valueOf() < Date.now();
```

Comment:
`getTime()` is more idiomatic:

```suggestion
        return new Date(event.endDate).getTime() < Date.now();
```